connectors/sources/onedrive.py

```diff
@@ -0,0 +1,223 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""OneDrive source module responsible to fetch documents from OneDrive.
+"""
+from datetime import datetime, timedelta
+from functools import cached_property
+from urllib import parse
+
+import aiohttp
+from aiohttp.client_exceptions import ClientResponseError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import (
+    CacheWithTimeout,
+    CancellableSleeps,
+    RetryStrategy,
+    iso_utc,
+    retryable,
+)
+
+RETRIES = 3
+RETRY_INTERVAL = 2
+BASE_URL = "https://graph.microsoft.com/v1.0/"
+ENDPOINTS = {"PING": "drives"}
```

Comment:
`me` endpoint cannot be used since it uses delegated permissions which requires user-level interaction in the oauth flow and we'll get following error:

```
        "code": "BadRequest",
        "message": "/me request is only valid with delegated authentication flow.",
```